### PR TITLE
Added local download example

### PR DESCRIPTION
**Pull request by dmonego**
Added a local download example that accepts a POST request. The serviceworker then sets the header to return a file as an attachment.

### CODE REVIEW

File: local-download/index.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+// Register the ServiceWorker
+navigator.serviceWorker.register('service-worker.js', {
+  scope: '.'
+}).then(function(registration) {
+  console.log('The service worker has been registered ', registration);
+  var content = document.getElementById("text");
```
Comment (delapuente) [inline local-download/index.js]: Remove the unused variables.

File: local-download/README.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+Allow a user to "download" a file that's been generated on the client side. 
+
+## Difficulty
+Beginner
+
+## Use Case
+Often it will be necessary to include a download feature in a single page application - for example, a draw program might want the ability to export as SVG, or a map 
+
+Because serviceworkers can respond to POST events, this allows us to attach the data to a POST operation. The service worker then pulls the data off of the request and returns it to the user as a file.
```
[the remark this answers is not in the document]
Comment (dmonego) [reply local-download/README.md]: GET is semantically better, but it limits you to the information you can put in the URL, so I don't think there are any real world examples where you would use it.

I've cleaned up this paragraph and pushed it with the rest of the suggested changes. Thanks for the review!

File: local-download/README.md [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+Allow a user to "download" a file that's been generated on the client side. 
+
+## Difficulty
+Beginner
+
+## Use Case
+Often it will be necessary to include a download feature in a single page application - for example, a drawing program might want the ability to export as SVG, or a bitmap format generated client side.
+
+
```
Comment (delapuente) [inline local-download/README.md]: Remove this blank line.

File: local-download/service-worker.js [hunk 4]
```diff
@@ -0,0 +1,16 @@
+// [Working example](/serviceworker-cookbook/local-download/).
+
+// Listen on fetch events for posts to download-file
+self.addEventListener('fetch', function(event) {
+    // If the request is going to the download-file endpoint, parse the post data and return a file. 
+    // This can be paired with a server side function with the same behavior as a fallback.
+    if(event.request.url.indexOf("download-file") !== -1) {
+        event.respondWith(event.request.formData().then(function (formdata){
```
Comment (delapuente) [inline local-download/service-worker.js]: Use two-spaces indentation, please.

File: local-download/index.html [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+  <style>
+    body {
+      font-family: monospace;
+      font-size: 1rem;
+      white-space: pre-wrap;
+    }
+  </style>
+</head>
+<body>
+  <form action="download-file" method="POST">
```
Comment (delapuente) [inline local-download/index.html]: Add labels to the inputs and make the UI a little bit more clear before merging.

**Pull request merged**